### ISSUE TITLE
20.03: `nixos-rebuild --upgrade`failing due to fontforge-20200314 build failure

### ISSUE THREAD

**Issue by codingkoi** [NixOS/nixpkgs#98475]
_Apologies if there was already a closed ticket about this. Issue search breaks when I try to look for closed tickets related to "fontforge" giving me the Github unicorn error page._

**Describe the bug**
I've been stuck unable to update my system because `fontforge-20200314` is failing to build. A truncated version of the build log:

```
CMake Error at /nix/store/acw9iism5wr6mxaq54h306bbxnl27n63-cmake-3.16.3/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:146 (message):
  Could NOT find Python3 (missing: Python3_EXECUTABLE Python3_LIBRARIES
  Python3_INCLUDE_DIRS Development Interpreter) (Required is at least version
  "3.3")
Call Stack (most recent call first):
  /nix/store/acw9iism5wr6mxaq54h306bbxnl27n63-cmake-3.16.3/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:393 (_FPHSA_FAILURE_MESSAGE)
  /nix/store/acw9iism5wr6mxaq54h306bbxnl27n63-cmake-3.16.3/share/cmake-3.16/Modules/FindPython/Support.cmake:2214 (find_package_handle_standard_args)
  /nix/store/acw9iism5wr6mxaq54h306bbxnl27n63-cmake-3.16.3/share/cmake-3.16/Modules/FindPython3.cmake:300 (include)
  cmake/PackageUtils.cmake:41 (find_package)
  CMakeLists.txt:137 (find_package_auto)


-- Configuring incomplete, errors occurred!
See also "/build/fontforge-20200314/build/CMakeFiles/CMakeOutput.log".
See also "/build/fontforge-20200314/build/CMakeFiles/CMakeError.log".
builder for '/nix/store/hr7lxgd2lwz0fk7b2h7g1a3ir1z5s5jh-fontforge-20200314.drv' failed with exit code 1
```

Seems like an issue with Python? I'm also not sure why this is being pulled in as it's not currently listed in my `/nix/store` so it must be a new dependency.

**To Reproduce**
I'm not sure how useful these steps will be given that it could be dependent on some state that I'm not sure how to convey. I would hope it's not given the statelessness of Nix, but :shrug:

1. run `sudo nixos-rebuild boot --upgrade`
2. see the error

**Expected behavior**
A successful system upgrade.

**Additional context**
Not sure what else is needed, but I'm happy to provide anything that might help.

**Notify maintainers**

@alyssais ? Sorry if this isn't relevant to your changes

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.
```
these paths will be fetched (0.05 MiB download, 0.28 MiB unpacked):
  /nix/store/d43v6bx7r6fcaq3fbbfd5mwh1f5s7rmg-bash-interactive-4.4-p23-dev
copying path '/nix/store/d43v6bx7r6fcaq3fbbfd5mwh1f5s7rmg-bash-interactive-4.4-p23-dev' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.50, NixOS, 20.03.2491.6a00eba02a3 (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.6`
 - channels(jeremy): `"unstable-20.09pre234194.8d05772134f"`
 - channels(root): `"nixos-20.03.3007.21d8e70a69f"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by giacomocariello** [comment]
Same here. In my case it's pulled in by libreoffice.

**Comment by codingkoi** [comment]
Oh that helps. I wasn't able to figure out which package was using it. I'll see if I can remove LibreOffice and try again to see if that works around it. I use that so infrequently.

**Comment by erictapen** [comment]
> I'm not sure how useful these steps will be given that it could be dependent on some state that I'm not sure how to convey

Good thing about Nix is that build errors should be independent of your systems *state*. But unfortunately they are dependent on your systems *configuration*.

The question is how your configuration evaluates to a broken fontforge. I tested both `21d8e70a69f704a6ab971b2d8265d40cc7bb69b1` (the Nixpkgs revision your channel pointed at when you opened the bug) and `b4db68ff563895eea6aab4ff24fa04ef403dfe14` (the latest revision of `nixos-20.03`) against `nixpkgs.fontforge`, `nixpkgs.fontforge-gtk` and `nixpkgs.fontforge-fonttools` (the three Nixpkgs attributes that are most likely meant by fontforge).

So as I see it the problem could be in lots of places. Lets start with your config:

Do you have anything in `nixpkgs.overlays` in your system configuration?

Also it could be interesting to find out what exactly depends on fontforge. For that you could run `nix-store -q --tree /run/current-system`, search for fontforge and identify any parents of it in that tree. Fontforge will very likely show up in that tree multiple times, as it is a reverse dependency for lots of packages. Some fonts are built with it, and fonts are everywhere…

**Comment by codingkoi** [comment]
Okay, thanks, those are places to start. It’s my work laptop so I’ll take a
look at it tomorrow since I already shut down for the day.

I do have a very small set of overlays so I’ll try removing them and see
how that goes.

Thanks for the command for finding how it’s getting pulled in. I went
looking for something like that but didn’t have much luck in my googling.

I’ll let you know what I find. Now that you mention overlays I kinda
suspect that.

**Comment by DiegoAlbertoTorres** [comment]
In my case, this was caused by the `monoid` font: https://github.com/NixOS/nixpkgs/blob/master/pkgs/data/fonts/monoid/default.nix

**Comment by erictapen** [comment]
@DiegoAlbertoTorres Thanks, I opened some PRs to fix `monoid` on `nixos-20.03`, `nixos-20.09` and `nixos-unstable`.

@codingkoi Any chance that you were also affected by the `monoid` package? I think it's likely and if its the case we could close this issue.

**Comment by codingkoi** [comment]
Yep, taking monoid out seems to have fixed it.
